src/components/DatePicker/index.android.js

```diff
@@ -8,87 +8,79 @@ import CONST from '../../CONST';
 import {propTypes, defaultProps} from './datepickerPropTypes';
 import styles from '../../styles/styles';
 
-class DatePicker extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isPickerVisible: false,
-        };
-
-        this.showPicker = this.showPicker.bind(this);
-        this.setDate = this.setDate.bind(this);
-    }
+function DatePicker({value, defaultValue, label, placeholder, errorText, containerStyles, disabled, maxDate, minDate, innerRef, onBlur, onInputChange}) {
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
 
     /**
      * @param {Event} event
      * @param {Date} selectedDate
      */
-    setDate(event, selectedDate) {
-        this.setState({isPickerVisible: false});
+    const setDate = (event, selectedDate) => {
+        setIsPickerVisible(false);
 
         if (event.type === 'set') {
             const asMoment = moment(selectedDate, true);
-            this.props.onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
+            onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
         }
-    }
+    };
 
-    showPicker() {
+    const showPicker = () => {
         Keyboard.dismiss();
-        this.setState({isPickerVisible: true});
-    }
+        setIsPickerVisible(true);
+    };
+
+    const setRef = (element) => {
+        if (!_.isFunction(innerRef)) {
+            return;
+        }
+        if (element && element.focus && typeof element.focus === 'function') {
+            let inputRef = {...element};
+            inputRef = {...inputRef, focus: showPicker};
+            innerRef(inputRef);
+            return;
+        }
```

Comment:
Is this something we can get rid of if we use `useImperativeHandle()`?  https://react.dev/reference/react/useImperativeHandle

Reply:
Hi @arthurmfgtab, please take a look at recently modified [forwardRef section in Contribution Guide](https://github.com/Expensify/App/blob/main/contributingGuides/TS_CHEATSHEET.md#forwardRef), where we've changed how `forwardRef`s are handled. Please let me know if anything worth mentioning is missing :D

Basically, we just skip additional layer in form of `innerRef` and pass refs directly to forwardRef.

### Code suggestion
Let's try to implement @marcaaron's suggestion:

```jsx
function DatePicker(props, outerRef) {
  const ref = useRef()

  ...

  const showPicker = () => {
      Keyboard.dismiss();
      setIsPickerVisible(true);
  };

  ...

  useImperativeHandle(outerRef, () => {
    return {
     ...ref.current,
     focus: () => {
      showPicker()
      // This was not included in original example but maybe it is needed?
      ref.current.focus?.()
   }}, [showPicker])

  ...

  return <TextInput ref={ref} ... />
}

export default React.forwardRef(DatePicker)
```

### Additional info
In this particular scenario we are using ref to extend behaviour of focus internally (closing keyboard and displaying date picker) without having to expose it to outside world, so I would suggest to simply use `onFocus` prop from RN `TextInput` for such functionality. It is more intuitive for most of React developers and can help avoid little bit of mental overhead :)

Additionally, please correct me if I am wrong, but there is no usage of the `DatePicker`'s ref across the code base, so we don't need to expose the ref at all.

Reply:
Thanks for that! ❤️ 

I think maybe we could still need this `focus()` method for some integration with the `Form` library - but agree with the points in general 👍